### PR TITLE
dotCMS/core#16523 Create dot-date-range

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        const tagsRenderExpected = `<dot-date-range label=\"Name:\" min=\"2019-01-01\" max=\"2019-12-31\" name=\"dateRange\" value=\"2019-05-01,2019-05-08\" hint=\"this is a hint\" required=\"true\" required-message=\"Required Name\" class=\"dot-valid dot-pristine dot-untouched dot-required hydrated\"><link rel=\"stylesheet\" href=\"https://cdn.jsdelivr.net/npm/flatpickr/dist/flatpickr.min.css\"><div class=\"dot-field__label\"><label for=\"dateRange\">Name:</label><span class=\"dot-field__required-mark\">*</span></div><input class=\"flatpickr-input\" id=\"dateRange\" required=\"\" type=\"text\" readonly=\"readonly\"><select><option value=\"0\">Date Presets</option><option value=\"-7\">Last Week</option><option value=\"7\">Next Week</option><option value=\"-30\">Last Month</option><option value=\"30\">Next Month</option></select><span class=\"dot-field__hint\">this is a hint</span></dot-date-range>`;
+        const dateValue = await input.getProperty('value');
+        expect(element.outerHTML).toBe(tagsRenderExpected);
+        expect(dateValue).toBe('2019-05-01 to 2019-05-08');
+        expect(element.classList.contains('dot-valid')).toBe(true);
```
Comment (hmoreras) [inline projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.e2e.ts]: All this 3 are not needed, are checked with the markup comparison.

File: projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        expect(element.classList.contains('dot-pristine')).toBe(true);
+        expect(element.classList.contains('dot-untouched')).toBe(true);
+    });
+
+    it('should be invalid', async() => {
+        element.setProperty('value', '');
+        await page.waitForChanges();
+        expect(element.classList.contains('dot-invalid')).toBe(true);
+        expect(element.classList.contains('dot-pristine')).toBe(true);
```
Comment (hmoreras) [inline projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.e2e.ts]: `expect ` for `dot-pristine` and `dot-untouched` not needed in this test.

File: projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        await page.waitForChanges();
+        const calendar = await page.find('.flatpickr-calendar');
+        expect(calendar.classList.contains('open')).toBeTruthy();
+        const days = await page.findAll('.flatpickr-day');
+        days[5].click();
+        days[8].click();
+        await page.waitForChanges();
+        const dateValue = await input.getProperty('value');
+        expect(dateValue).toBe('2019-05-03,2019-05-06');
+        expect(element.classList.contains('dot-valid')).toBe(true);
```
Comment (hmoreras) [inline projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.e2e.ts]: are these needed here ?

File: projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        const today = dt.toISOString().split('T')[0];
+        dt.setDate(dt.getDate() + preset);
+        const future = dt.toISOString().split('T')[0];
+
+        const dateValue = await input.getProperty('value');
+        expect(dateValue).toBe(`${today},${future}`);
+    });
+
+    it('should clear value, set pristine and untouched  when input set reset', async () => {
+        element.callMethod('reset');
```
Comment (hmoreras) [inline projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.e2e.ts]: add await.

File: projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
+import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
```
Comment (hmoreras) [inline projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.e2e.ts]: missing test for invalid scenario ?

File: projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.tsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+import { getClassNames, getOriginalStatus, updateStatus, getTagLabel, getTagHint, getErrorClass, getTagError } from '../../utils';
+import flatpickr from 'flatpickr';
+
+@Component({
+    tag: 'dot-date-range',
+    styleUrl: 'dot-date-range.scss'
+})
+export class DotDateRangeComponent {
+    @Element() el: HTMLElement;
+    @Prop({ mutable: true }) value = '';
```
Comment (hmoreras) [inline projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.tsx]: add documentation for props ?

File: projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.tsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+            label: 'Next Week',
+            time: 7
+        }, {
+            label: 'Last Month',
+            time: -30
+        }, {
+            label: 'Next Month',
+            time: 30
+        }
+    ];
```
Comment (freddyucv) [inline projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.tsx]: Why this is a Prop?, I think it should be in another file

File: projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.tsx [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+        return !(this.required && !(this.value && this.value.length));
+    }
+
+    private setValue(selectedDates, _dateStr, _instance): void {
+        this.value =
+            selectedDates && selectedDates.length === 2
+                ? `${selectedDates[0].toISOString().split('T')[0]},${
+                      selectedDates[1].toISOString().split('T')[0]
+                  }`
+                : '';
```
Comment (freddyucv) [inline projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.tsx]: the value not should be a String, we should keep it in a better type and turn it into a String in emitValueChange

**Pull request merged**